cpp/src_prims/distance/pairwise_distance_base.cuh

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <raft/linalg/contractions.cuh>
+#include <raft/linalg/norm.cuh>
+
+namespace MLCommon {
+namespace Distance {
+
+/**
+ * @brief Device class for L2 expanded, unexpanded distance metrics.
```

Comment:
nitpick: this is supposed to be the base class for all distance metrics, right?

Reply:
yes, will correct this.